frame/collective/src/lib.rs

```diff
@@ -694,29 +551,236 @@ decl_module! {
+	#[pallet::storage]
+	#[pallet::getter(fn proposal_of)]
+	pub type ProposalOf<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Identity,
+		T::Hash,
+		T::Proposal,
+		OptionQuery,
+	>;
+
+	/// Votes on a given proposal, if it is ongoing.
+	#[pallet::storage]
+	#[pallet::getter(fn voting)]
+	pub type Voting<T: Config<I>, I: 'static = ()> = StorageMap<
+		_,
+		Identity,
+		T::Hash,
+		Votes<T::AccountId, T::BlockNumber>,
+		OptionQuery
+	>;
+
+	/// Proposals so far.
+	#[pallet::storage]
+	#[pallet::getter(fn proposal_count)]
+	pub type ProposalCount<T: Config<I>, I: 'static = ()> = StorageValue<_, u32, ValueQuery>;
+
+	/// The current members of the collective. This is stored sorted (just by value).
+	#[pallet::storage]
+	#[pallet::getter(fn members)]
+	pub type Members<T: Config<I>, I: 'static = ()> = StorageValue<_, Vec<T::AccountId>, ValueQuery>;
+
+	/// The prime member that helps determine the default vote behavior in case of absentations.
+	#[pallet::storage]
+	#[pallet::getter(fn prime)]
+	pub type Prime<T: Config<I>, I: 'static = ()> = StorageValue<_, T::AccountId, OptionQuery>;
+
+	/// (default) if not.
+	#[pallet::storage]
+	#[pallet::getter(fn upgraded_pallet_prefix)]
+	pub(super) type UpgradedPalletPrefix<T: Config<I>, I: 'static = ()> =
+		StorageValue<_, bool, ValueQuery>;
+
+	#[pallet::genesis_config]
+	pub struct GenesisConfig<T: Config<I>, I: 'static = ()>{
+		pub members: Vec<T::AccountId>,
+		pub phantom: sp_std::marker::PhantomData<(T, I)>,
+	}
+
+	#[cfg(feature = "std")]
+	impl<T: Config<I>, I: 'static> Default for GenesisConfig<T, I> {
+		fn default() -> Self {
+			Self {
+				members: Default::default(),
+				phantom: Default::default(),
+			}
+		}
+	}
+
+	#[pallet::genesis_build]
+	impl<T: Config<I>, I: 'static> GenesisBuild<T, I> for GenesisConfig<T, I> {
+		fn build(&self) {
+			<UpgradedPalletPrefix<T, I>>::put(true);
+			Pallet::<T, I>::initialize_members(&self.members);
+		}
+	}
+}
+
+mod migrations {
+	use super::*;
+
+	pub fn migrate_to_new_pallet_prefix<T: Config<I>, I: 'static>() -> frame_support::weights::Weight {
+
+		let new_name = T::PalletInfo::name::<Pallet<T, I>>()
+			.expect("Fatal Error Invalid PalletInfo name")
+			.as_bytes();
+
+		if new_name == b"Collective" {
+			return 0
+		}
+
+		move_storage_from_pallet(b"Proposals", b"Collective", new_name);
+		move_storage_from_pallet(b"ProposalOf", b"Collective", new_name);
+		move_storage_from_pallet(b"Voting", b"Collective", new_name);
+		move_storage_from_pallet(b"ProposalCount", b"Collective", new_name);
+		move_storage_from_pallet(b"Members", b"Collective", new_name);
+		move_storage_from_pallet(b"Prime", b"Collective", new_name);
+
+		T::BlockWeights::get().max_block
+	}
+}
+
+#[cfg(feature = "std")]
+impl<T: Config<I>, I: 'static> GenesisConfig<T, I> {
+	/// Direct implementation of `GenesisBuild::build_storage`.
+	///
+	/// Kept in order not to break dependency.
+	pub fn build_storage(&self) -> Result<sp_runtime::Storage, String> {
```

Comment:
actually we can deprecate them
```suggestion
	#[deprecated(note = "use [`GenesisBuild::build_storage`] instead")]
	pub fn build_storage(&self) -> Result<sp_runtime::Storage, String> {
```